docs/lang/articles/contribution/contributor_guide.md

```diff
@@ -4,296 +4,324 @@ sidebar_position: 1
-  with tags starting with **the first letter capitalized**, e.g.,
-  `[Metal]`, `[Vulkan]`, `[IR]`, `[Lang]`, `[CLI]`. These PRs will be
-  [highlighted in the release note](https://github.com/taichi-dev/taichi/blob/master/misc/make_changelog.py)
-  for end-users, therefore it is important to make sure your PR title is
-  effective in telling what your PR does.
-- Other PRs (underlying development or intermediate implementation)
-  should use tags with **everything in lowercase letters**, e.g.,
-  `[metal]`, `[vulkan]`, `[ir]`, `[lang]`, `[cli]`.
-- Because of the way the release changelog is generated, there
-  should be **at most one capitalized tag** in a PR title to prevent
-  duplicate PR highlights. For example,
-  `[GUI] [Mac] Support modifier keys` ([#1189](https://github.com/taichi-dev/taichi/pull/1189))
-  is an improper tag choice, and we
-  should have used `[gui] [Mac] Support modifier keys in GUI` instead.
-  Please capitalize the tag that is the *most* relevant to the PR.
+We do appreciate all kinds of contributions, yet we do not intend to expose the details of every PR to the end-users. Therefore the changelog helps distinguish *what the user should know* from *what the*
+
+*developers are doing*. This is done by **capitalizing PR tags**:
+
+- PRs with visible or notable features to the users should be marked with tags starting with **a capitalized initial**, for example`[Metal]`, `[Vulkan]`, `[IR]`, `[Lang]`. These PRs are [highlighted in the release note](https://github.com/taichi-dev/taichi/blob/master/misc/make_changelog.py) for the end users, therefore it is important that your PR title be self-explanatory.
+
+- Other PRs (underlying development or intermediate implementation) should use tags **in lowercase letters**, for example`[metal]`, `[vulkan]`, `[ir]`, `[lang]`.
+
+- Because of the way the release changelog is generated, there should be **at most one capitalized tag** in a PR title to prevent duplicate PR highlights. For example, `[GUI] [Mac] Support modifier keys` ([#1189](https://github.com/taichi-dev/taichi/pull/1189))is an improper tag choice, and we should have used `[gui] [Mac] Support modifier keys in GUI` instead.
+
+Please capitalize the most relevant tag to your PR.
+
 :::
 
-## Testing
+### **Use continuous integration**
 
-Tests should be added to [tests/](https://github.com/taichi-dev/taichi/blob/master/tests/). We
-have both Python tests and C++ tests.
+- Continuous Integration (CI) will **build** and **test** your commits in a specific PR in multiple environments.
 
-### Python tests
+- CI is triggered each time you push a commit to an open PR.
+
+- A tick on the left-hand side of a commit hash means that CI passes, whilst a cross means that CI fails.
+
+### **PR reviewing & merging**
+
+- Please try to follow these tips from Google:
+  - [Code Health: Understanding Code In](https://testing.googleblog.com/2018/05/code-health-understanding-code-in-review.html)
```

Comment:
Typo: the original content only had two list items. It should have been

* Code Health: Understanding Code In Review
* Code Health: Respectful Reviews == Useful Reviews

https://github.com/taichi-dev/taichi/blob/9f0d9f505cfdb9aec6317fb394ef88d01e9cb2d2/docs/lang/articles/contribution/contributor_guide.md?plain=1#L141-L147